### PR TITLE
Bugzilla-1254143: Switch to pip hash checking mode

### PR DESCRIPTION
**Pull request by bqbn**
Added hashes for modules whose versions are pinned with "==". Need to
break the existing requirement files into 2 types: with and without
hashes. This is because according to the doc,

> Hash verification is an all-or-nothing proposition. Specifying a
> --hash against any requirement not only checks that hash but also
> activates a global hash-checking mode.

For example, for those modules whose versions are not pinned with "==",
or those that are installed from VCS directly, if they're mixed in the
same file with modules that use "--hash" option, the install fails with
errors similar to the following:

> In --require-hashes mode, all requirements must have their versions
> pinned with ==. These do not:
>     simplejson>=3.0.5 from https://pyrepo.stage.mozaws.net/olympia/simplejson-3.8.1.tar.gz
>     (from -r compiled_without_hash.txt (line 16))

or

> The editable requirement django-queryset-transform from
> git+https://github.com/jbalogh/django-queryset-transform@a1ba6ae41bd86f5bb9ff66fb56614e0fafe6e022#egg=django-queryset-transform
> (from -r compiled_without_hash.txt (line 17)) cannot be
> installed when requiring hashes, because there is no single file to
> hash.

### CODE REVIEW

**Comment by bqbn** [comment]
@jasonthomas r?
This should be merged after https://github.com/mozilla-services/svcops/pull/946 is merged in order not to break the rpm build.

**Comment by EnTeQuAk** [comment]
We're installing `addons-server` into the environment too - https://travis-ci.org/mozilla/addons-server/jobs/122280415#L255 - we should probably split that something like https://gist.github.com/EnTeQuAk/1ca81c7c1d4f44ab185987059f4d57ca and make sure we're installing it separately from the requirements-files on deployment too.

**Comment by bqbn** [comment]
Yep, I'll look into this.

File: requirements/compiled.txt [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+                 --hash=sha256:7929e35163f316751fe2d33dd1405729e3fa709f94fc1a29989efe8271705f39
+Pillow==2.8.1 --hash=sha256:835e7e0b54256c035c718d5e2f5ebce94822915176a2cc7680f402b967a288a0 \
+              --hash=sha256:8760c118a0215eba163f7782110e7efcdbb15f8a7321f3f61c5ac0dbbb12c996 \
+              --hash=sha256:fb426a905cabe89df722ddc8141fd696b2983d9553714d1750b11f2eff28184b
 
-# For the addon validator, including C speedups.
-# 'simplejson>=3.0.5' is used so that docker hub doesn't remove the
-# pre-installed version as that causes a segfault. Other use-cases
-# need '>' so that they can install a version that exists in PyPi.
-simplejson>=3.0.5
```
Comment (jasonthomas) [inline requirements/compiled.txt]: We should investigate if we can pin this to the version in pyrepo. Based on the commit history I believe simplejson==2.3.2 was causing a segfault.

Comment (bqbn) [inline requirements/compiled.txt]: @jasonthomas I agree. As we talked offline, I filed issue https://github.com/mozilla/addons-server/issues/2336 to lock down the simplejson issue.

**Comment by bqbn** [comment]
Merged the latest master and resolved the conflict in requirements/prod.txt.

Can someone take a look and merge as early as possible so I don't have to maintain this branch locally? :)

**Comment by jasonthomas** [comment]
r+

**Pull request merged**